nomad/state/state_store.go

```diff
@@ -1187,15 +1187,14 @@ func (s *StateStore) deleteJobFromPlugin(index uint64, txn *memdb.Txn, job *stru
 	plugins := map[string]*structs.CSIPlugin{}
 
 	for _, a := range allocs {
-		tg := job.LookupTaskGroup(a.TaskGroup)
```

Comment:
Hi @michaeldwan I think the error we made here is that this line should have read:

```go
tg := a.job.LookupTaskGroup(a.TaskGroup)
```

It seemed strange to me that a job can not have the taskgroup for one of its own allocs, but my colleague @notnoop has pointed out to me that the alloc can have a view of the job that's stale relative to the job we've pulled from the state store, and that's why we're running into this panic.

Reply:
That's roughly what we think happened too. We were actually working on a network issue that impacted consul but nomad seemed healthy through it.

I can make that change here.

Reply:
Gah, I typoed my example. It's `a.Job` (capital letter) 😊

Reply:
I just made the change. There's a few other places LookupTaskGroup is being called but they appear to get both the job and alloc from the same snapshot and should be okay.

Reply:
I made the same typo and pushed before waiting for tests in vagrant to run :)